knotx-core/knotx-template-engine/src/test/java/com/cognifide/knotx/engine/placeholders/UriInfoHelperTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Knot.x - Reactive microservice assembler - Templating Engine Verticle
+ *
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.knotx.engine.placeholders;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import com.cognifide.knotx.api.UriInfo;
+
+@RunWith(Parameterized.class)
+public class UriInfoHelperTest {
+
+    @Parameterized.Parameters(name = "{index}: {0}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {{"/a/b", "/a/b", null, null, null},
+                {"/a/b.html", "/a/b", null, "html", null},
```

Comment:
I like the those tests!
However i am not a fan of regular expressions... 

Why aren't we using [JAVA URI](https://docs.oracle.com/javase/7/docs/api/java/net/URI.html) class?

That would also address my other concerns about URI path